libs/vscode/mcp/src/lib/mcp-server.ts

```diff
@@ -21,232 +22,313 @@ import { getOutputChannel } from '@nx-console/vscode-output-channels';
 import { getTelemetry } from '@nx-console/vscode-telemetry';
 import {
   getGitDiffs,
-  getNxMcpPort,
   isInVSCode,
   sendMessageToAgent,
   vscodeLogger,
 } from '@nx-console/vscode-utils';
-import express from 'express';
+import express, { Request, Response } from 'express';
 import { commands, ProgressLocation, tasks, window } from 'vscode';
 
-export interface McpServerReturn {
-  server: NxMcpServerWrapper;
-  app: express.Application;
-  server_instance: ReturnType<express.Application['listen']>;
-  clearKeepAliveInterval: () => void;
-}
+export class McpWebServer {
+  private static instance: McpWebServer;
+  public static get Instance() {
```

Comment:
Just a thought. does there need to be a private constructor to make sure no one outside can accidentally instantiate this mcp server and get an instance if not via `.getInstance()`? 🤔